reV/supply_curve/supply_curve.py

```diff
@@ -259,6 +259,42 @@ def _parse_trans_table(trans_table):
 
         return trans_table
 
+    @staticmethod
+    def _map_trans_capacity(trans_sc_table):
+        """
+        Map SC gids to transmission features based on capacity. For any SC
+        gids with capacity > the maximum transmission feature capacity, map
+        SC gids to the feature with the largest capacity
+
+        Parameters
+        ----------
+        trans_sc_table : pandas.DataFrame
+            Table mapping supply curve points to transmission features.
+
+        Returns
+        -------
+        trans_sc_table : pandas.DataFrame
+            Updated table mapping supply curve points to transmission features
+            based on maximum capacity
+        """
+        mask = trans_sc_table['capacity'] <= trans_sc_table['max_cap']
+        over_max = trans_sc_table.loc[~mask].copy()
+        trans_sc_table = trans_sc_table.loc[mask]
+        if over_max.size:
+            msg = ("The following SC points have a capacity that "
+                   "exceeds the maximum transmission feature capacity "
+                   "{}, and will be mapped to features with the max "
+                   "capacity:\n{}"
+                   .format(over_max['sc_gid'].unique(),
+                           np.max(over_max['max_cap'])))
+            logger.warning(msg)
+            warn(msg)
+            for _, grp_df in over_max.groupby(['sc_gid', 'trans_gid']):
+                idx_max = grp_df['max_cap'].argmax()
+                trans_sc_table = trans_sc_table.append(grp_df.iloc[[idx_max]])
```

Comment:
I think there's an unnecessary amount of bracketing going on here.

Reply:
You'd think right! turns out you can't append a series to a dataframe. Doing iloc[[]] returns a dataframe

Reply:
I didnt know that! cool.